cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/TranslatedFunction.qll

```diff
@@ -209,8 +210,15 @@ class TranslatedFunction extends TranslatedRootElement, TTranslatedFunction {
       (
         // Only generate the `Unwind` instruction if there is any exception
         // handling present in the function.
-        exists(TryStmt try | try.getEnclosingFunction() = func) or
+        exists(TryStmt try | try.getEnclosingFunction() = func)
+        or
+        exists(MicrosoftTryStmt try | try.getEnclosingFunction() = func)
+        or
         exists(ThrowExpr throw | throw.getEnclosingFunction() = func)
+        or
+        exists(FunctionCall call | call.getTarget().(ThrowingFunction).mayThrowException(true) |
+          call.getEnclosingFunction() = func
```

Comment:
Why do we only generate an `Unwind` instruction when the function _must_ throw? I am aware that we currently only have `ThrowingFunction` functions that _must_ throw, but at some point we will probably model functions that _may_ throw, and they also need an `Unwind` instruction, right?

Reply:
Currently it doesn't make much sense to me to do this in the case of `mayThrowException(false)`, as in that case we will not output an exception edge directly after the call.

Reply:
Hm. I thought we would output both an exceptional edge and a goto edge in that case? 🤔

Reply:
In that case, it needs to change, and I've misunderstood the relevant code in that case. Let me have another look.

Reply:
You were right. Fixed in https://github.com/github/codeql/pull/17421/commits/5f4fee07802b97848390df23748ac026b72c787f.